### PR TITLE
Update MSON namespace to the latest spec

### PR DESCRIPTION
**Pull request by zdne**
This PR updates the MSON namespace to reflect the changes in base spec done in #14 

---

cc @kylef @smizell @klokane

### CODE REVIEW

File: namespaces/mson-namespace.md [hunk 1]
```diff
@@ -289,13 +277,10 @@ Enumeration type. Exclusive list of possible elements. The elements in content's
...
-                            "content": {
-                                "key": {
-                                    "element": "string",
-                                    "content": "state"
-                                }
-                            }
+                            "element": "string",
+							"meta": {
+								"name": "state"
+							}
```
Comment (smizell) [inline namespaces/mson-namespace.md]: Is formatting off here?

File: namespaces/mson-namespace.md [hunk 2]
```diff
@@ -303,13 +288,10 @@ Enumeration type. Exclusive list of possible elements. The elements in content's
...
-                            "content": {
-                                "key": {
-                                    "element": "string",
-                                    "content": "province"
-                                }
-                            }
+                            "element": "string",
+							"meta": {
+								"name": "province"
+							}
```
Comment (smizell) [inline namespaces/mson-namespace.md]: Formatting?

File: namespaces/mson-namespace.md [hunk 3]
```diff
@@ -337,13 +319,10 @@ Using the `ref` element to reference an the content of an element.
...
-            "content": {
-                "key": {
-                    "element": "string",
-                    "content": "id"
-                }
-            }
+            "element": "string",
+			"meta": {
+				"name": "id"
+			}
```
Comment (smizell) [inline namespaces/mson-namespace.md]: Formatting?

File: namespaces/mson-namespace.md [hunk 4]
```diff
@@ -237,20 +234,14 @@ Enumeration type. Exclusive list of possible elements. The elements in content's
...
-                    },
-                    "content": null
-                }
-            }
+            "element": "number",
+            "meta": {
+                "name": "id"
+            },
+            "attributes": {
+                "default": 0
```
Comment (smizell) [inline namespaces/mson-namespace.md:248]: In the definition, it says the value of "default" is an element. Does this go along with that?

**Comment by smizell** [comment]
@zdne A couple of formatting issues maybe and [this](https://github.com/refractproject/refract-spec/pull/15/files?diff=split#r29731771). Other than that, good to go!

File: namespaces/mson-namespace.md [hunk 5]
```diff
@@ -2,20 +2,37 @@
...
-1. [Number Type](#number-type-number-element)
-1. [Array Type](#array-type-array-element)
-1. [Object Type](#object-type-object-element)
-1. [Property Type](#property-type-property-element)
-1. [Enum Type](#enum-type-mson-element)
-
-# About this Document
+## Content
+
+<!-- TOC depth:3 withLinks:1 updateOnSave:0 -->
```
Comment (kylef) [inline namespaces/mson-namespace.md:7]: Is this automatically generated somehow? Would love to hear how this magic works.

Comment (zdne) [inline namespaces/mson-namespace.md:7]: @kylef yes

https://atom.io/packages/markdown-toc

there is a sublime text version too, very handy

cc @smizell

File: namespaces/mson-namespace.md (hunk 1 above)
Comment (zdne) [inline namespaces/mson-namespace.md]: Strange. That was some tabs / space mismatch – looked OK in my editor. Fixing it right now

File: namespaces/mson-namespace.md (hunk 4 above)
Comment (zdne) [inline namespaces/mson-namespace.md:248]: Good catch! In reality this should be the same type as element's `content`. I am afraid the MSON syntax is a bit short here. Going to update `MSON Element` definition.

**Comment by zdne** [comment]
Updated.

File: namespaces/mson-namespace.md [hunk 6]
```diff
@@ -75,8 +91,14 @@ Note: In MSON DOM _Nested Member Types_ _Type Section_ is the `content` of the e
...
-    - `sample` (*T*) - Alternative sample value for _Member Types_
-    - `default` (*T*) - Default value for _Member Types_
+    - `sample` - Alternative sample value for _Member Types_
+
+		The type of of `sample` attribute MUST match the type of element's `content`.
+
+    - `default` - Default value for _Member Types_
+
+		The type of of `default` attribute MUST match the type of element's `content`.
+
```
Comment (smizell) [inline namespaces/mson-namespace.md:101]: This is just a thought, but could this concept ever be useful in MSON? That is, do you ever see a need to be able to express the type of one value must match the type of another, where that other value may be an enum?

Just a thought. Not an issue here.

File: namespaces/mson-namespace.md (hunk 4 above)
Comment (pksunkara) [inline namespaces/mson-namespace.md:248]: @zdne Is this going to be in another PR?

Comment (zdne) [inline namespaces/mson-namespace.md:248]: @pksunkara ~~Yup~~

Comment (zdne) [inline namespaces/mson-namespace.md:248]: @pksunkara not sure what you mean – can you please clarify?

Comment (pksunkara) [inline namespaces/mson-namespace.md:248]: Sorry. I see that you have updated the `MSON Element`.
